src/reactive/duplicity.py

```diff
@@ -40,6 +41,37 @@
 config = hookenv.config()
 
 
+class PipPackageInstallError(RuntimeError):
+    """Error during package installation with pip."""
+
+    pass
+
+
+def install_in_system_python(package_to_install):
+    """Install dependency in system python.
+
+    The charm use subprocess to call duplicity,
+    which is installed by apt.
+    So the azure-storage-blob has to be installed n system level python
+    See https://duplicity.nongnu.org/vers7/duplicity.1.html#sect10
+    There is no maintained package in apt repositories,
+    so we have to install it with pip.
+    """
+    command = ["sudo", "pip", "install", package_to_install]
+    output = subprocess.run(
+        ["sudo", "pip", "install", package_to_install],
+        stdout=subprocess.PIPE,
+        stderr=subprocess.PIPE,
+    )
+    if output.returncode != 0:
+        raise PipPackageInstallError(
```

Comment:
Nothing specifically handles this exception, so it's probably not worth creating a new exception type?  It might also be neater to log the information rather than letting it be dumped into the traceback.  Maybe something more like:

```
try:
 subprocess.check_*(...)
except calledprocesserror as e:
  log.error(command failed)
  log.error(e.stderr)
  raise
```

Reply:
i believe it will be helpful in future, because while working for this feature i face so many issue while installing a package. So i believe it would be good to have, and unit tests will also be much aligned for future to check for the exception raised.

Reply:
Ok, that's fair :+1: